### ISSUE TITLE
 gcc 11.2 warnings

### ISSUE THREAD

**Issue by ghost** [openwall/john#5150]
Below a non-SIMD build.
gcc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)

```
In file included from truecrypt_fmt_plug.c:61:
truecrypt_fmt_plug.c: In function 'crypt_all._omp_fn.0':
pbkdf2_hmac_sha512.h:55:23: error: 'lens[0]' may be used uninitialized in this function [-Werror=maybe-uninitialized]
   55 |         for (i = 0; i < KL; i++) {
      |                     ~~^~~~
truecrypt_fmt_plug.c:481:21: note: 'lens[0]' was declared here
  481 |                 int lens[INNER_BATCH_MAX_SZ];
      |                     ^~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:1556: truecrypt_fmt_plug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:190: default] Error 2

Exited with code exit status 2
```

**Comment by AlekseyCherepanov** [comment]
Most probably I introduced that. Anyway I am familiar with this code. I'll fix. Thanks!

**Comment by AlekseyCherepanov** [comment]
That's a false warning:
```c
	int i, inner_batch_size = 1;
[...]
		int lens[INNER_BATCH_MAX_SZ];
		int j;

		for (j = 0; j < inner_batch_size; ++j) {
			lens[j] = strlen((char *)key_buffer[i+j]);
[...]
			pbkdf2_sha512((const unsigned char*)keys[0], lens[0], psalt->salt, 64, psalt->num_iterations, keys[0], sizeof(keys[0]), 0);
```

In non-simd build `inner_batch_size` is 1 always, so I can replace it with a constant. It solves the warning.

Interesting that `pbkdf2_ripemd160` and `pbkdf2_whirlpool` do not yield such warning.

**Comment by AlekseyCherepanov** [comment]
Before the fix, this warning would happen on gcc 8.3.0 and 10.2.1. So I could catch it testing my code for #5127. But this warning is specific for non-simd + openmp build. Without openmp, it does not occur.
